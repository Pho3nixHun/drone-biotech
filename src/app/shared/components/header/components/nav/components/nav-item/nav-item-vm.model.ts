export interface NavItemVM {
  titleKey: string;
}
