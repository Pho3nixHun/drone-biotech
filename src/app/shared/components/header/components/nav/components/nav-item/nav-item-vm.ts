export interface NavItemVM {
  href: string;
}
