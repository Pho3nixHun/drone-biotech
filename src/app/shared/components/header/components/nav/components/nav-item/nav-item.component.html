@if (vm(); as vm) {
  <a [href]="vm.href" class="sm:hover:text-gray-400 duration-150 cursor-pointer">
    <ng-content />
  </a>
}
