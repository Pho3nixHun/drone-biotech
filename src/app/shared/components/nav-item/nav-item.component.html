<a class="sm:hover:text-gray-400 sm:cursor-pointer duration-150">
  <ng-content />
</a>
