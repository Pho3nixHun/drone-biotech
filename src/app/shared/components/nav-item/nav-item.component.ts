import { Component } from '@angular/core';

@Component({
  selector: 'app-nav-item',
  standalone: true,
  imports: [],
  templateUrl: './nav-item.component.html',
})
export class NavItemComponent {

}
