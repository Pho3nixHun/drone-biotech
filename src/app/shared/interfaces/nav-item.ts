export interface NavItem {
  link: string;
  text: string;
}
