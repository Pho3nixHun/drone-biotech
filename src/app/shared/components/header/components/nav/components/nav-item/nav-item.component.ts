import { Component, input } from '@angular/core';
import { NavItemVM } from './nav-item-vm.model';
import { TranslocoModule } from '@jsverse/transloco';

@Component({
  selector: 'app-nav-item',
  standalone: true,
  imports: [TranslocoModule],
  templateUrl: './nav-item.component.html',
})
export class NavItemComponent {
  vm = input<NavItemVM | null>(null, { alias: 'vm' });
}
