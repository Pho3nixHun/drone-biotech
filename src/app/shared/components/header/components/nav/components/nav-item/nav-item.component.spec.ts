import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NavItemComponent } from './nav-item.component';
import { getTranslocoModule } from 'transloco-testing.module';
import { NavItemVM } from './nav-item-vm.model';

describe('NavItemComponent', () => {
  let component: NavItemComponent;
  let fixture: ComponentFixture<NavItemComponent>;
  let compiled: HTMLElement;
  const vm: NavItemVM = { titleKey: 'Home' };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NavItemComponent, getTranslocoModule()],
    }).compileComponents();

    fixture = TestBed.createComponent(NavItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should get the input and visualize it', () => {
    fixture.componentRef.setInput('vm', vm);
    fixture.detectChanges();
    expect(compiled.innerText).toContain(vm.titleKey);
  });
});
