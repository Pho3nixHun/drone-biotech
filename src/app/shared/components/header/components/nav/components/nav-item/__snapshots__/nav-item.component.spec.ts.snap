// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItemComponent should not render the template when the VM is not provided 1`] = `
<div
  id="root1"
  ng-version="18.2.7"
>
  <app-nav-item>
    <!--container-->
  </app-nav-item>
</div>
`;

exports[`NavItemComponent should render the template with the provided VM and ng-content 1`] = `
<div
  id="root2"
  ng-version="18.2.7"
>
  <app-nav-item
    ng-reflect-vm="[object Object]"
  >
    <a
      class="sm:hover:text-gray-400 duration-150 cursor-pointer"
      href="home"
    >
      Home
    </a>
    <!--container-->
  </app-nav-item>
</div>
`;
